.github/workflows/test-scheduler-prop.yml

```diff
@@ -0,0 +1,54 @@
+name: Scheduler property testing
+on:
+  pull_request:
+    paths:
+      - 'scheduler/**'
+      - 'nomad/structs/**'
+  push:
+    branches:
+      - main
+      - release/**
+    paths:
+      - 'scheduler/**'
+      - 'nomad/structs/**'
+
+jobs:
+  property-tests:
+    timeout-minutes: 20
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+        with:
+          cache: ${{ contains(runner.name, 'Github Actions') }}
+          go-version-file: .go-version
+          cache-dependency-path: '**/go.sum'
+
+      - name: Run property tests
+        run: |
+          go test -v -cover ./scheduler/reconciler -rapid.checks=100000 -run PropTest
```

Comment:
Note for reviewers: as it turns out, the `-rapid` flags won't get detected properly if we try to use a package spec like `./scheduler/...`, so we need a separate line for each package we want to cover. Also, running 100k iterations doesn't take very long but I figure we may want to keep this short for the moment just until we've got some useful work for them to do, and then we can crank it up to 1M or whatever.